### PR TITLE
Merge PropfindJob with LsColJob

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

File: src/gui/quotainfo.cpp [hunk 1]
```diff
@@ -99,14 +99,14 @@ void QuotaInfo::slotCheckQuota()
...
+    _job = new PropfindJob(account, account->davUrl(), quotaBaseFolder(), 0, this);
     _job->setProperties({ QByteArrayLiteral("quota-available-bytes"), QByteArrayLiteral("quota-used-bytes") });
-    connect(_job.data(), &PropfindJob::result, this, &QuotaInfo::slotUpdateLastQuota);
+    connect(_job.data(), &PropfindJob::directoryListingIterated, this, &QuotaInfo::slotUpdateLastQuota);
     connect(_job.data(), &AbstractNetworkJob::networkError, this, &QuotaInfo::slotRequestFailed);
     _job->start();
 }
 
-void QuotaInfo::slotUpdateLastQuota(const QMap<QString, QString> &result)
+void QuotaInfo::slotUpdateLastQuota(const QString &, const QMap<QString, QString> &result)
```
Comment (fmoc) [inline src/gui/quotainfo.cpp:109]: If the string isn't used, why add it to the signature?

File: src/gui/sharedialog.h [hunk 2]
```diff
@@ -52,7 +52,7 @@ class ShareDialog : public QDialog
     QSize minimumSizeHint() const override;
 
 private slots:
-    void slotPropfindReceived(const QMap<QString, QString> &result);
+    void slotPropfindReceived(const QString &, const QMap<QString, QString> &result);
```
Comment (fmoc) [inline src/gui/sharedialog.h:55]: Wouldn't it be better style to adapt the connections?

Alternatively we might add a second signature with a FIXME that calls the original one.

File: src/libsync/networkjobs.cpp [hunk 3]
```diff
@@ -294,73 +290,35 @@ bool LsColXMLParser::parse(const QByteArray &xml, QHash<QString, qint64> *sizes,
...
-LsColJob::LsColJob(AccountPtr account, const QUrl &url, const QString &path, QObject *parent)
+PropfindJob::PropfindJob(AccountPtr account, const QUrl &url, const QString &path, int depth, QObject *parent)
     : AbstractNetworkJob(account, url, path, parent)
+    , _depth(depth)
 {
     // Always have a higher priority than the propagator because we use this from the UI
     // and really want this to be done first (no matter what internal scheduling QNAM uses).
     // Also possibly useful for avoiding false timeouts.
     setPriority(QNetworkRequest::HighPriority);
+    Q_ASSERT(depth == 0 || depth == 1);
```
Comment (fmoc) [inline src/libsync/networkjobs.cpp:301]: This was introduced in 3b3531f77753d1c72909d057e0f79058c0044ae7, but I don't see why it's important. You should probably add a comment.

File: src/libsync/networkjobs.h [hunk 4]
```diff
@@ -75,11 +75,11 @@ class OWNCLOUDSYNC_EXPORT LsColXMLParser : public QObject
     void finishedWithoutError();
 };
 
-class OWNCLOUDSYNC_EXPORT LsColJob : public AbstractNetworkJob
+class OWNCLOUDSYNC_EXPORT PropfindJob : public AbstractNetworkJob
 {
     Q_OBJECT
 public:
-    explicit LsColJob(AccountPtr account, const QUrl &url, const QString &path, QObject *parent = nullptr);
+    explicit PropfindJob(AccountPtr account, const QUrl &url, const QString &path, int depth = 1, QObject *parent = nullptr);
```
Comment (fmoc) [inline src/libsync/networkjobs.h:82]: Since we allow only two values, it might be easier (and safer) to introduce one or two named constructors.

**Review by fmoc — CHANGES_REQUESTED** [review]

File: src/gui/quotainfo.cpp (hunk 1 above)
Comment (TheOneRing) [reply src/gui/quotainfo.cpp:109]: Because thats how connects work

File: src/gui/sharedialog.h (hunk 2 above)
Comment (TheOneRing) [reply src/gui/sharedialog.h:55]: An additional signal would need to be emitted.
How would you change the connect?

File: src/gui/quotainfo.cpp (hunk 1 above)
Comment (fmoc) [reply src/gui/quotainfo.cpp:109]: https://github.com/owncloud/client/pull/10181#discussion_r993405244

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/13389/6/1

<details>


</details>

File: src/gui/sharedialog.h (hunk 2 above)
Comment (fmoc) [reply src/gui/sharedialog.h:55]: There's the option of connecting a lambda which calls the slot correctly. But that's less efficient.

**Review by fmoc — APPROVED** [review]
Approving. Please make sure it builds, though.

**Pull request merged**